Add `Newtype` derivation to `Block`

XertroV opened the pull request:
Since we don't have lenses on `Block` fields a `Newtype` instance let's us use `unwrap` and the like.

Not sure if we had this before or not, but noticed it was missing now.

blinky3713 commented:
we don't really use that anywhere (transaction, receipt, etc). maybe we should include instances for all such objects if we have it for one

XertroV commented:
> maybe we should include instances for all such objects

Yup - agreed. It's not used in the library but I definitely use it - I prefer `block <- unwrap <$> runWeb3 ...` to `(Block block) <- runWeb3 ...` and the like. Just makes it a bit more flexible.

I'll add in `Newtype` derivations to other types tomorrow.

XertroV commented:
@blinky3713 - Added the rest of newtype derivations

Other changes:
* removed some lines to make spacing of derivations more consistent
* added Eq derivation for TransactionOptions (a bunch of other ones had it and I don't see why it shouldn't have an Eq instance)

Only types we don't have newtype derivations for are stuff like HexStrings (where there are good reasons for not letting people `wrap` and `unwrap`, etc)

XertroV commented:
@blinky3713 - FYI added some test for newtype derivations to make sure they aren't removed (I think I added at least some of them before that got removed once).

Anyway, I'm not sure sure how best to test for derivations - basically here I just rely on the typechecker and throw away the result of any web3 computations (so it doesn't matter if a node is running or not)

If you have any better ideas please let me know.

blinky3713 commented:
🎉
